src/PremiumLib.cpp

```diff
@@ -0,0 +1,429 @@
+/**
+    This plugin can be used for common player customizations
+ */
+
+#include "AccountMgr.h"
+#include "ScriptMgr.h"
+#include "Player.h"
+#include "Config.h"
+#include "Chat.h"
+#include "Language.h"
+#include "CharacterDatabase.h"
+#include "LoginDatabase.h"
+#include "DatabaseWorkerPool.h"
+#include "MySQLConnection.h"
+#include "PremiumLib.h"
+
+/* Premium character interaction.
+     * These functions will allow modules to access the new generic
+     * premium tables designed to help modules that interact with
+     * players and characters without the need to create other tables.
+*/
+
+PremiumLibData PremiumLibManager::GetAccountPremiumLevel(uint64 accountID)
+{
+    QueryResult result = LoginDatabase.PQuery("SELECT premium_level, expiration_date, FROM_UNIXTIME(expiration_date) FROM premium_account WHERE account_id = %u", accountID);
```

Comment:
so if we use this function in another module, it will return the date in human readable time and not unixtime?

I'm not sure about this.

Of course, for the native commands it's good, but to use in another module, I don't think it's good to manipulate dates like "3rd of april, 2020".

What about adding an optional argument to the functions and if enabled, it will use "FROM_UNIXTIME" (just set it in a variable) ?

Note: maybe i'm wrong and didn't understand the code, quite possible as I review it quickly and you know that I don't do CPP

Reply:
It will return the premium level, expiration date in unixtime and the same expiration date in human readable time.

Reply:
true my bad :P 

what about two optional arguments

`PremiumLibData PremiumLibManager::GetAccountPremiumLevel(uint64 accountID, return_unix_date = true, return_readable_date = false)`

so we decide what to return depending on the option when calling the function
well it's just a detail, it won't change much at all

Reply:
Well, it's the same, the struct PremiumLibData is already built and it will be the same size filled or not filled. Also, readable data is a string and unixtime is int, so is not compatible to return in the same method.